.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+## <a name="commit"></a> Commit Message Guidelines
+
+We have very precise rules over how our git commit messages can be formatted.  This leads to **more
+readable messages** that are easy to follow when looking through the **project history**. For full contributiong guidelines visit
+the [Contributors Guide](https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-Commits) on the EdgeX Wiki
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+The footer should contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages/) if any.
+
+Example 1:
+```
+build(makefile): add docker support
+```
+```
+fix(logging): Initialize remote logging properly
+
+Previously remote logging failed due to improper initialization of the logging client. This commit fixes the initialization to properly support remote logging.
+
+Closes: #123
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the CI/CD pipline or build system or external dependencies (example scopes: travis, jenkins, makefile)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix
+* **perf**: A code change that improves performance
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, etc)
+* **test**: Adding missing tests or correcting existing tests
+
+### Scope
+The scope should be the name of the module or package affected (as perceived by the person reading the changelog generated from commit messages.
+
+The following is the list of suggested scopes:
```

Comment:
The scopes for ci-management would be different. We can remove all of the modules/packages scopes.

Suggested ci-management scopes:
* **jenkins-config**
* **jjb**
* **packer**
* **pipeline**

@ernestojeda @jamesrgregg @soda480 @MightyNerdEric Some feedback on this is appreciated.